Improbable/Stdlib/Improbable.Stdlib/WorkerConnection.cs

```diff
@@ -178,61 +174,58 @@ public void StopSendingMetrics()
-                {
-                    throw new AuthenticationException($"Error received while retrieving a Player Identity Token: {value.Value.Status.Detail}");
-                }
+            if (!value.HasValue)
+            {
+                throw new AuthenticationException("Error received while retrieving a Player Identity Token: null result");
+            }
 
-                return value.Value.PlayerIdentityToken;
+            if (value.Value.Status.Code != ConnectionStatusCode.Success)
+            {
+                throw new AuthenticationException($"Error received while retrieving a Player Identity Token: {value.Value.Status.Detail}");
             }
+
+            return value.Value.PlayerIdentityToken;
         }
 
-        private static List<LoginTokenDetails> GetDevelopmentLoginTokens(string host, ushort port, bool useInsecureConnection, string workerType, string pit)
+        private static IEnumerable<LoginTokenDetails> GetDevelopmentLoginTokens(string host, ushort port, bool useInsecureConnection, string workerType, string pit)
         {
-            using (var tokens = DevelopmentAuthentication.CreateDevelopmentLoginTokensAsync(host, port,
+            using var tokens = DevelopmentAuthentication.CreateDevelopmentLoginTokensAsync(host, port,
                 new LoginTokensRequest
                 {
                     PlayerIdentityToken = pit,
                     WorkerType = workerType,
-                    UseInsecureConnection = useInsecureConnection,
-                }))
-            {
-                var value = tokens.Get();
+                    UseInsecureConnection = useInsecureConnection
+                });
 
-                if (!value.HasValue)
-                {
-                    throw new AuthenticationException("Error received while retrieving Login Tokens: null result");
-                }
+            var value = tokens.Get();
 
-                if (value.Value.Status.Code != ConnectionStatusCode.Success)
-                {
-                    throw new AuthenticationException($"Error received while retrieving Login Tokens: {value.Value.Status.Detail}");
-                }
+            if (!value.HasValue)
+            {
+                throw new AuthenticationException("Error received while retrieving Login Tokens: null result");
+            }
 
-                if (value.Value.LoginTokens.Count == 0)
-                {
-                    throw new Exception("No deployment returned for this project.");
-                }
+            if (value.Value.Status.Code != ConnectionStatusCode.Success)
+            {
+                throw new AuthenticationException($"Error received while retrieving Login Tokens: {value.Value.Status.Detail}");
+            }
 
-                return value.Value.LoginTokens;
+            if (value.Value.LoginTokens.Count == 0)
+            {
+                throw new Exception("No deployment returned for this project.");
```

Comment:
Nit
I'm not sure if this should be an exception. Its totally expected to have no deployments running.

Reply:
Agreed, I've removed this, it's up to client code to decide what to do